source/codegen/templates/library.cpp.mako

```diff
@@ -44,6 +45,10 @@ ${service_class_prefix}Library::${service_class_prefix}Library() : shared_librar
 %>\
   function_pointers_.${method_name} = reinterpret_cast<${method_name}Ptr>(shared_library_.get_function_pointer("${c_name}"));
 % endfor
+% if 'SetRuntimeEnvironment' in service_helpers.filter_api_functions(functions, only_mockable_functions=False):
+
+  this->SetRuntimeEnvironment(nidevice_grpc::NIDEVICE_GRPC_ORIGINALFILENAME.c_str(), nidevice_grpc::NIDEVICE_GRPC_FILEVERSION.c_str(), "", "");
```

Comment:
Maybe I need to do a try, catch in case this errors?

Reply:
Rather than adding specific logic for a specific function, it might be better to define this requirement in the metadata and use that to add any extra "initialization functions" so that this is extensible to other functions in the future. I'll defer to @reckenro or @astarche on that since they know more about the code generation, though.

Reply:
Like Marcos, I was expecting something like:

```c++
  if (function_pointers_.SetRuntimeEnvironment) {
    function_pointers_.SetRuntimeEnvironment(environment, environmentVersion, reserved1, reserved2);
  }
```

But I agree that try/catch is not the end of the world.

I'm always a fan of adding metadata conventions. But I think there's already a strong convention in the drivers to implement/use `SetRuntimeEnvironment` in this way and that it's implemented similarly in `nimi-python` so I'm happy for the team to use their judgement on it.

Is it possible that we'd want to automatically add `SetRuntimeEnvironment` even if drivers don't declare it in their metadata? If the convention is strong enough, it can be better to have it be global.

Reply:
I actually just added that check.
Let me know if you think I need to document the reason for the call.

Reply:
>Rather than adding specific logic for a specific function, it might be better to define this requirement in the metadata and use that to add any extra "initialization functions" so that this is extensible to other functions in the future. I'll defer to @reckenro or @astarche on that since they know more about the code generation, though.

I'll stick with the YAGNI (You Ain't Gonna Need It) principle and avoid overcomplicating it, since I haven't been told other wise.